OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -1606,6 +1606,110 @@ public static DreamValue NativeProc_sorttext(DreamObject instance, DreamObject u
+            {
+                return new DreamValue(0);
+            }
+
+            if(start < 0)
+            {
+                start = Math.Max(start + text.Length + 1, 1);
+            }
+
+            int result = 0;
+            while(start <= text.Length)
+            {
+                if(text.AsSpan(start - 1, 1).IndexOfAny(needles) > -1)
+                {
+                    result++;
+                }
+                else
+                {
+                    break;
+                }
+                start++;
+            }
+            return new DreamValue(result);
+        }
+
+        [DreamProc("spantext_char")]
+        [DreamProcParameter("Haystack", Type = DreamValueType.String)]
+        [DreamProcParameter("Needles", Type = DreamValueType.String)]
+        [DreamProcParameter("Start", Type = DreamValueType.Float, DefaultValue = 1)]
+        public static DreamValue NativeProc_spantext_char(DreamObject instance, DreamObject usr, DreamProcArguments arguments) {
+            //if any arguments are bad, return 0
+            if (!arguments.GetArgument(0, "Haystack").TryGetValueAsString(out var text))
+            {
+                return new DreamValue(0);
+            }
+            if (!arguments.GetArgument(1, "Needles").TryGetValueAsString(out var needles))
+            {
+                return new DreamValue(0);
+            }
+            if (!arguments.GetArgument(2, "Start").TryGetValueAsInteger(out var start)) //0 is not valid, 
+            {
+                return new DreamValue(0);
+            }
+            if(start > text.Length)
+            {
+                return new DreamValue(0);
+            }
+            StringInfo textStringInfo = new StringInfo(text);
+            
+            if(start < 0)
+            {
+                start = Math.Max(start + textStringInfo.LengthInTextElements + 1, 1);
+            }
+
+            int result = 0;
+            
+            TextElementEnumerator needlesElementEnumerator = StringInfo.GetTextElementEnumerator(needles);
+            
+            TextElementEnumerator textElementEnumerator = StringInfo.GetTextElementEnumerator(text);
+            textElementEnumerator.Reset();
+
+            while(textElementEnumerator.MoveNext()) 
+            {
+                if(textElementEnumerator.ElementIndex >= start - 1)
+                {
+                    bool found = false;
+                    needlesElementEnumerator.Reset();
+                    while(needlesElementEnumerator.MoveNext()) //lol O(N*M)
+                    {
+                        if(textElementEnumerator.Current.Equals(needlesElementEnumerator.Current))
+                        {
+                            result++;
+                            found=true;
+                            break;
+                        }
+                    }
+                    if(!found)
+                    {
+                        break;
+                    }
+                }
```

Comment:
This `if` is unnecessary if you set the starting index like  in the previous suggestion.
```suggestion
                bool found = false;
                needlesElementEnumerator.Reset();
                
                //lol O(N*M)
                while (needlesElementEnumerator.MoveNext()) {
                    if (textElementEnumerator.Current.Equals(needlesElementEnumerator.Current)) {
                        result++;
                        found = true;
                        break;
                    }
                }

                if (!found) {
                    break;
                }
```